### ISSUE TITLE
Handle config option "useslash" when renaming pages

### ISSUE THREAD

**Issue by Saruspete** [michitux/dokuwiki-plugin-move#153]
Hello,

Upon renaming, it seems the rewriting of existing links does not handle the [config "useslash"](https://www.dokuwiki.org/config:useslash) properly. I can see only one usage of this parameter in https://github.com/michitux/dokuwiki-plugin-move/blob/master/helper/handler.php#L142 but it does not seems to be working correctly.

Instead of using a fixed ":" in the code, it would be better to assign this separator to a variable / constant and use it instead.

Thanks for the work :)

**Comment by michitux** [comment]
So you are suggesting when the `useslash` is on, all adapted links should use `/` as separator instead of `:`? Or are you saying that links with `/` as namespace separator are not or not properly changed? I'm asking because I personally prefer to have `:` in the wiki text even though I also like to have `/` in URLs.

**Comment by splitbrain** [comment]
@saruspete thank you for giving us an regexp tutorial.

@michitux I'd vote for won't implement. I agree with you that useslash is mostly cosmetics for the URL. The canonical way to write a pageid is using colons.

**Comment by Saruspete** [comment]
@splitbrain so you'd rather provide a feature to end user (useslash) and limits its use and go against end-user choice, for the sole sake of cosmetics ?

(If you take this code as a regex tutorial, you may come play with us at regexcrossword.com )

**Comment by michitux** [comment]
@Saruspete Note also that even if the regex was only applied to links themselves, what you propose is unable to handle relative links or links that are not valid page ids but become a matching id once they have been cleaned (like `Foo Bar` which becomes `foo_bar`).

**Comment by Saruspete** [comment]
@michitux Indeed, makes sense. I was missing all these internals. Thanks a lot for the explanation !

When you say configuration option, is it for the whole wiki, in the configuration panel ? 
Or just for a default value for a checkbox on the move form (to be displayed only if `useslash` is enabled) ?

If it's too many changes for such a "one-shot" feature, I can live with a sed on the data files.
